src/utils/update/psc.to.beneficial.owner.type.mapper.ts

```diff
@@ -157,3 +158,22 @@ const natureOfControlTypeMap = new Map<string, NatureOfControlType>([
   [natureOfControl.SIGNIFICANT_INFLUENCE_OR_CONTROL_AS_FIRM, NatureOfControlType.SIGNIFICANT_INFLUENCE_OR_CONTROL]
 ]);
 
+export const mapBoPrivateAddress = (boData: BeneficialOwnersPrivateData, ch_reference: string) => {
+  if (!boData?.boPrivateData?.length) {
+    return;
+  }
+
+  for (const private_bo_data of boData.boPrivateData) {
+    if (private_bo_data.hashedId === ch_reference) {
+      const usualResidentialAddress = (private_bo_data.usualResidentialAddress
+                                      && checkAddressValues(private_bo_data.usualResidentialAddress));
+      return usualResidentialAddress
+        ? mapBOMOAddress(private_bo_data.usualResidentialAddress)
+        : mapBOMOAddress(private_bo_data.principalAddress);
+    }
+  }
+};
+
+const checkAddressValues = (address: PrivateAddress) => {
```

Comment:
elsewhere we test for keys with:
Object.keys(address || {}).length !== 0
which would make check on line 168 simpler too.
so could be just just:
return Object.keys(private_bo_date.usualResidentialAddress || {}).length !== 0 ?
       mapBOMOAddress(private_bo_data.usualResidentialAddress) : mapBOMOAddress(private_bo_data.principalAddress);

Reply:
seems to be returning undefined in the tests :(

Reply:
for both residential and principal addresses*

Reply:
It could be that they keys are there but values undefined. We should know which address to use by the type of MO. 
So should we pass in an indicator that allows us to select the right address to use. Rather than checking for which address we have. 
For individual BOs it's the usualResidentialAddress 
For BO Other or BO Gov it's the principalAddress (registeredOfficeAddress)
Then we would not have to check for null.
For MOs - 
MO indivdual - use usualResidentialAddress
MO Corporate - use principalAddress
but we can do that while or after merging

Reply:
yeah, I could try now if you want, but if we want to get it in asap then maybe on one of the other tickets, let me know!

Reply:
Happy to try that after.

Reply:
got your changes in, please have a look whenever you're free! @AndreKramer7654